src/tests/math/test_linspace.f90

```diff
@@ -0,0 +1,245 @@
+program test_linspace
+    use stdlib_error, only: check
+    use stdlib_kinds, only: sp, dp, int16, int8
+    use stdlib_math, only: linspace 
+
+    implicit none
+
+    integer :: iunit
+    logical :: warn = .false.
+
+    ! Testing linspace.
+    !
+    ! For single and double precision, check if the beginning and end values are properly calculated
+    ! and make sure that the size of the result is as expected.
+    !
+    !
+
+
+    open(newunit=iunit, file="test_linspace_log.txt", status="unknown") ! Log the results of the functio
+
+    write(iunit,*) "Writing to unit #: ", iunit
+
+    call test_linspace_sp
+    call test_linspace_dp
+    call test_linspace_neg_index ! Make sure that when passed a negative index the result is an empty array
+    call test_linspace_cmplx
+    call test_linspace_cmplx_2
+    call test_linspace_cmplx_3
+    call test_linspace_cmplx_sp
+    call test_linspace_cmplx_sp_2
+    call test_linspace_int16
+    call test_linspace_int8
+
+    close(unit=iunit)    
+
+
+contains
+
+    subroutine test_linspace_sp
+
+        integer :: n = 20
+        real(sp) :: start = 1.0_sp
+        real(sp) :: end = 10.0_sp
+
+        real(sp), dimension(:), allocatable :: x
+
+        x = linspace(start, end, n)
+
+        call check(x(1) == start, msg="Initial value of array is not equal to the passed start parameter", warn=warn)
```

Comment:
Could you check the result against the true answer?

Reply:
I am now checking the starting and ending values of the result to be bit-exact with the expected initial values. 
I am also checking to make sure that the true difference between adjacent elements of `res` is within a `TOLERANCE` (0.001%) of the `expected_interval`